flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/PurgingTrigger.java

```diff
@@ -44,39 +44,36 @@ private  PurgingTrigger(Trigger<T, W> nestedTrigger) {
 	@Override
 	public TriggerResult onElement(T element, long timestamp, W window, TriggerContext ctx) throws Exception {
 		TriggerResult triggerResult = nestedTrigger.onElement(element, timestamp, window, ctx);
-		switch (triggerResult) {
-			case FIRE:
-				return TriggerResult.FIRE_AND_PURGE;
-			case FIRE_AND_PURGE:
-				return TriggerResult.FIRE_AND_PURGE;
-			default:
-				return TriggerResult.CONTINUE;
+		if (triggerResult.isFire()) {
+			return TriggerResult.FIRE_AND_PURGE;
+		} else if (triggerResult.isPurge()) {
+			return TriggerResult.PURGE;
```

Comment:
Couldn't we simplify it even more like the following:
`return triggerResult.isFire() ? TriggerResult.FIRE_AND_PURGE : triggerResult;`